Update razzle to use vswhere (#13)

amweiss opened the pull request:
Instead of listing out all the locations for MSBuild, use the vswhere tool to find the location and add that directory to that path.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/microsoft/Terminal?pullRequest=606) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/Terminal?pullRequest=606) before we can accept your contribution.<br/><br/>:x: amweiss [sign now](https://cla.opensource.microsoft.com/microsoft/Terminal?pullRequest=606)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/microsoft/Terminal?pullRequest=606) it.</sub>

DHowett-MSFT commented:
I can't figure out where in the nuget docs it specifies that `.nuget` is a good and sane location, but I'm glad it works.

heaths commented on tools/razzle.cmd:
```diff
@@ -84,6 +18,20 @@ set OPENCON=%OPENCON_TOOLS:~0,-7%
+)
+
+rem Add path to MSBuild Binaries
+for /f "usebackq tokens=*" %%i in (`%VSWHERE% -latest -requires Microsoft.Component.MSBuild -find MSBuild\**\Bin\MSBuild.exe`) do set MSBUILD=%%i
```
You should add `-products *` if you want to also support Visual Studio Build Tools. By default, vswhere only supports finding Community, Professional, and Enterprise - our traditional VS SKUs.

zadjii-msft commented on tools/razzle.cmd:
```diff
@@ -84,6 +18,20 @@ set OPENCON=%OPENCON_TOOLS:~0,-7%
+for /f "usebackq delims=" %%I in (`dir /b /aD /o-N /s "%~dp0..\packages\vswhere*"`) do (
+    for /f "usebackq delims=" %%J in (`where /r "%%I" vswhere.exe 2^>nul`) do set VSWHERE=%%J
+)
+
```
Could we add an error message if VsWhere isn't found? Similar to what used to be on L69-72?

amweiss replied:
Done

zadjii-msft commented on tools/razzle.cmd:
```diff
@@ -84,6 +18,20 @@ set OPENCON=%OPENCON_TOOLS:~0,-7%
+
+rem Add path to MSBuild Binaries
+for /f "usebackq tokens=*" %%i in (`%VSWHERE% -latest -products * -requires Microsoft.Component.MSBuild -find MSBuild\**\Bin\MSBuild.exe`) do set MSBUILD=%%i
+
```
If VsWhere can't find MsBuild, what happens here? Does it try to treat an error message as a path to MsBuild?

amweiss replied:
Nope, and now it bails out when it fails per your other request

zadjii-msft commented on README.md:
```diff
@@ -120,8 +120,8 @@ git submodule update --init --recursive
-nuget restore OpenConsole.sln
-msbuild OpenConsole.sln
+.\tools\razzle.cmd
+msbuild
```
_mega nit_: could we change L124 to `bcz` while we're here?

amweiss replied:
done, and I had to add a change to bcz so it could handle spaces in MSBUILD

amweiss commented:
> Couple questions, but I'm not gonna block this over them

I can tackle these after work today, or on another PR later if you want this in sooner.

zadjii-msft commented:
@amweiss I'm in no rush - my unread inbox is expanding at just about the rate I reply to things :P

amweiss commented:
Ha, I expected it would be worse than that by now 😁. Anyway, changes made. I did expand the scope a bit when I changed the readme to use `bcz` since it failed when the `MSBUILD` path had a space, so I fixed that by wrapping it in quotes in `bcz`.

miniksa commented:
/azp run

Daniel15 commented on tools/razzle.cmd:
```diff
@@ -84,6 +20,34 @@ set OPENCON=%OPENCON_TOOLS:~0,-7%
+
+rem Find vswhere
+rem from https://github.com/microsoft/vs-setup-samples/blob/master/tools/vswhere.cmd
+for /f "usebackq delims=" %%I in (`dir /b /aD /o-N /s "%~dp0..\packages\vswhere*" 2^>nul`) do (
```
Is there a reason this needs to be a batch file rather than using PowerShell? I imagine this could be clearer than the obscure `dir` command if it used PowerShell.

amweiss replied:
I was just following what was already there.

zadjii-msft replied:
Because not everyone likes Powershell. Our dev team is split between users of Powershell and cmd, so we have scripts for both.
